x/crosschain/keeper/evm_hooks.go

```diff
@@ -256,10 +258,28 @@ func (k Keeper) ParseZRC20WithdrawalEvent(ctx sdk.Context, log ethtypes.Log) (*z
 		return nil, err
 	}
 
-	_, found := k.fungibleKeeper.GetForeignCoins(ctx, event.Raw.Address.Hex())
+	coin, found := k.fungibleKeeper.GetForeignCoins(ctx, event.Raw.Address.Hex())
 	if !found {
 		return nil, fmt.Errorf("ParseZRC20WithdrawalEvent: cannot find foreign coin with contract address %s", event.Raw.Address.Hex())
 	}
+	chainID := coin.ForeignChainId
+	if common.IsBitcoinChain(chainID) {
+		if event.Value.Cmp(big.NewInt(0)) <= 0 {
+			return nil, fmt.Errorf("ParseZRC20WithdrawalEvent: invalid amount %s", event.Value.String())
+		}
+		btcChainParams, err := common.GetBTCChainParams(chainID)
+		if err != nil {
+			return nil, err
+		}
+		addr, err := btcutil.DecodeAddress(string(event.To), btcChainParams)
+		if err != nil {
+			return nil, fmt.Errorf("ParseZRC20WithdrawalEvent: invalid address %s: %s", event.To, err)
+		}
+		_, ok := addr.(*btcutil.AddressWitnessPubKeyHash)
+		if !ok {
+			return nil, fmt.Errorf("ParseZRC20WithdrawalEvent: invalid address %s (not P2WPKH address)", event.To)
+		}
```

Comment:
IMO it would be better to create a separate function for verification and add unit tests for it

Reply:
indeed. We'll refactor this later.